src/codegen/mod.rs

```diff
@@ -646,7 +646,20 @@ impl RetrieveType for Expression {
     }
 }
 
+/// Increment an expression by some value.
+///
+/// Assumes
 impl Expression {
+    pub(crate) fn add_u32(&self, other: &Expression) -> Self {
```

Comment:
```suggestion
    pub(crate) fn add_u32(self, other: Expression) -> Self {
```
We end up consuming the expressions we add, so a suggestion would be to move the values instead of receiving the reference. We can leave the choice wether to clone the expression to where this function is called.

Reply:
If you implement this, don't forget to remove the clones.